arcgis-ios-sdk-samples/Edit data/Add features with contingent values/ContingentValuesTableViewController.swift

```diff
@@ -0,0 +1,313 @@
+                self.selectedBufferSize = nil
+            }
+            // Remove the options view controller after selection.
+            self.navigationController?.popViewController(animated: true)
+        }
+        // Set the options view controller's title and present it.
+        optionsViewController.title = "Protection"
+        self.show(optionsViewController, sender: self)
+    }
+    
+    /// Get the minimum and maximum values of the possible buffer sizes.
+    func showBufferSizeOptions() {
+        guard let feature = feature else { return }
+        // Get the contingent value results using the feature and field.
+        let contingentValueResult = featureTable?.contingentValues(with: feature, field: "BufferSize")
+        guard let bufferSizeGroupContingentValues = contingentValueResult?.contingentValuesByFieldGroup["BufferSizeFieldGroup"] as? [AGSContingentRangeValue] else { return }
+        // Set the minimum and maximum possible buffer sizes.
+        let minValue = bufferSizeGroupContingentValues[0].minValue as! Int
+        let maxValue = bufferSizeGroupContingentValues[0].maxValue as! Int
+        bufferSizes = Array(minValue...maxValue)
+        // Select the buffer size if it is the only option available.
+        if bufferSizes?.count == 1 {
+            selectedBufferSize = bufferSizes?[0]
+        }
+        // Update the picker view.
+        bufferSizePickerView.reloadAllComponents()
+    }
+    
+    func createFeature(with status: AGSCodedValue) {
+        // Get the contingent values definition from the feature table.
+        let contingentValuesDefinition = featureTable?.contingentValuesDefinition
+        // Load the contingent values definition.
+        contingentValuesDefinition?.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else if let feature = self.featureTable?.createFeature() as? AGSArcGISFeature {
+                // Create a feature from the feature table and set the initial attribute.
+                feature.attributes["Status"] = status.code
+                self.feature = feature
+            }
+        }
+    }
+
+    /// Ensure that the selected values are a valid combination.
+    func validateContingency() {
+        guard let featureTable = featureTable, let feature = feature else { return }
+        // Validate the feature's contingencies.
+        let contingencyViolations = featureTable.validateContingencyConstraints(with: feature)
+        if contingencyViolations.isEmpty {
+            // If there are no contingency violations in the array, the feature is valid and ready to add to the feature table.
+            doneBarButtonItem.isEnabled = true
+        } else {
+            // Present an alert if there are contingency violations.
+            self.presentAlert(title: "", message: "Invalid contingent values")
```

Comment:
```suggestion
            presentAlert(title: "", message: "Invalid contingent values")
```
It may be too much to add here, but do you think printing out the field group of the contingency is helpful?

Reply:
Sure, I can add that. I think it'd be helpful for troubleshooting if users enter their own data.